src/util.rs

```diff
@@ -268,34 +268,27 @@ impl RequestBuilderExt for RequestBuilder {
     }
 }
 
-type BoxFuture<T> = std::pin::Pin<Box<dyn std::future::Future<Output = T> + Send + 'static>>;
-
-// TODO: Avoid boxing?
 pub(crate) trait ResponseExt: Sized {
```

Comment:
1. This change is unrelated to the title about wasm support, thus should not be included here.
2. Using async-fn-in-trait will [bump MSRV to 1.75](https://caniuse.rs/features/return_position_impl_trait_in_trait). I don't see an obvious reason for trading that. Network operation itself is so costly that `Box`ing costs relatively nothing, thus this change will not improve performance either.

Reply:
On wasm `reqwest::Response` is not `Send`, thus incompatible with the `Box<Future>` as declared.

Reply:
I think just removing the `Send` requirement would work, but I saw the `TODO` comment and went straight to use `async fn` in `impl`. Gonna try removing `Send` to avoid bumping MSRV.